### ISSUE TITLE
`getNode` returns undefined after HMR runs

### ISSUE THREAD

**Issue by totalhack** [formkit/formkit#251]
### Reproduction

N/A

### Describe the bug

This is unfortunately not reproducible in the playground as it depends on scenarios you can't hit there, but I've run into this same issue in two different instances now. 

In one case I see this any time Vite does a HMR on my starter project that uses the `stepPlugin` from [this](https://github.com/formkit/formkit/issues/63) multi-step form example. I've reproduced it using @andrew-boyd's example [as-is](https://formkit.link/da55ac3f0088bffc6b00625dd979ed5b) and with my own [modified version](https://formkit.link/0f50968e6fa3183ddf359b3ec1983119) of the plugin.  In both cases the call to `getNode` in the code that tries to walk nodes to trigger validation returns `undefined`. This does *not* happen with a full page refresh, only after a dev HMR.

The other case I am seeing the same thing is when trying to package and load this as a Vue component built with Vite's library mode. This is experimental as well but just looking for a good way to make this reusable across other projects, so I was able to package and deploy it without issue, and it loads the initial form without issue, but as soon as I click the Next button and it tries to walk the nodes we hit the `undefined` response from `getNode` again (this time without any HMR). Note that I've validated the node name it's looking for is correct.

That project is here: https://github.com/totalhack/formkit-multistep

You can do an npm install of `formkit-multistep` into a project, add `import { FormKitMultiStep } from 'formkit-multistep'` into your App.vue, and then pass it a `schema` prop as is done in my modified version of the multi-step form example -- see [this](https://github.com/totalhack/formkit-multistep/blob/master/src/App.vue) for a dev example within the formkit-mulistep repo, but note that that won't reproduce it because its loading `FormKitMultiStep` from the local component instead of the npm module. Perhaps this is some ESM issue like what's described in [this](https://github.com/formkit/formkit/issues/115
) issue?

Is there perhaps some pre-req step required to ensure whatever `getNode` is doing under the hood has knowledge of all nodes, which perhaps is not running (or getting cleared out) on HMR? Not sure if the ESM thing is a separate issue.

EDIT: I stepped through this a bit with the debugger and confirmed the global `registry` is empty after HMR. Perhaps something needs to recreate/reinit the nodes on a hot reload?

### Environment

• OS: Mac OS
• Browser: Chrome / Firefox
• Version: Latest

**Comment by justin-schroeder** [comment]
I'm definitely going to need more to reproduce this. My guess is the hot module reloading is breaking it. If you provide a reproduction repository we can take a crack at it.

**Comment by totalhack** [comment]
I believe you can reproduce using this: https://github.com/totalhack/formkit-multistep

Clone, npm install, npm run dev. The dev App.vue will run and you can then trigger/observe the issue in the browser console by making a code change to cause HMR and trying to click the Next button.

**Comment by justin-schroeder** [comment]
Perfect, thanks for the repro. Yeah interesting, looks like HMR is messing with it somehow. If I was to guess, I'm betting that HMR mounts the new component before unmounting the old one, so when the old one unmounts it takes the reference out of the registry with it.

Not sure what the best way around that is, but its worth looking into for sure.

**Comment by totalhack** [comment]
Btw, this seems to happen even without HMR if you `npm install formkit-multistep` and import FormKitMultiStep from that...getNode returns undefined when clicking Next on the first try in that case. Not sure if it's the same underlying issue though.

**Comment by totalhack** [comment]
Hey @justin-schroeder, just wanted to let you know I at least have a very lightly tested workaround that addresses the HMR situation, though not by fixing any underlying issue with `getNode`. 

Basically instead of using `getNode`, in the useSteps.js plugin I tracked the node objects for the steps. This is updated in version 0.0.2 of https://github.com/totalhack/formkit-multistep so I can keep playing around without hitting the HMR issue at least.

```javascript
// in stepPlugin:
steps[node.name].node = node;

// in the code that looks for step nodes to validate
const node = steps[currentStep].node
```

Surprisingly I still get the undefined node issue when loading this from an npm-installed component even with the new approach. Any known issues I should be aware of with packaging formkit as part of a vue component? I added more debug prints (into v0.0.3) and when the component is loaded from the npm module the useSteps plugin init never seems to get called. I know the code is there from inspecting the dist build files, and get some prints on clicking Next from functions within useSteps before it hits the undefined node issue, but the plugin is never initialized so the latter is not surprising.

My guess is this has something to do with the "$plugins" value on the schema not finding anything, even though I've verified that value exists on the [mergedData](https://github.com/totalhack/formkit-multistep/blob/master/src/components/FormKitMultiStep.vue#L13) value being passed into `<FormKitSchema :schema="schema" :data="mergedData" />`. Apologies if this is getting a little hairy/specific to my use case, but it's still unclear to me if I'm doing something wrong or hitting a bug in the schema stuff. Is there any debug capability I can enable on the `FormKitSchema` node such that it will dump out what context it has available on init?

```
const schema = [
  {
    $cmp: 'FormKit',
    props: {
      type: 'form',
      id: 'form',
      onSubmit: '$submit("https://httpbin.org/post")',
      plugins: '$plugins',
      actions: false, 
    },
```
